TEST.md

```diff
@@ -0,0 +1,192 @@
+
+```bash
+$ bin/test-run `which conduit` specialtest
+```
+
+That will create multiple namespaces in your Kubernetes cluster:
+
+```bash
+$ kubectl get ns | grep specialtest
+specialtest               Active    4m
+specialtest-egress-test   Active    2m
+specialtest-get-test      Active    1m
+...
+```
+
+You can also test a locally-built version of the `conduit` CLI. Note, however,
+that this requires that you build the corresponding Conduit docker images and
+publish them to a docker registry that's accessible from the Kubernetes cluster
+where you're planning on running the tests. As a result, local testing mostly
+applies to [minikube](https://github.com/kubernetes/minikube), since you can
+build the images directly into minikube's local docker registry, as described
+below.
+
+To test your current branch on minikube, first build all of the Conduit images
+in your minikube environment, by running:
+
+```bash
+$ DOCKER_TRACE=1 bin/mkube bin/docker-build
+```
+
+That command also outputs the corresponding `conduit` binaries into the
+`target/cli` directory, and you can use the `bin/conduit` script to load those
+binaries when running tests. To run tests using your local binary, run:
+
+```bash
+$ bin/test-run `pwd`/bin/conduit
+```
+
+That will create multiple namespaces in your Kubernetes cluster:
+
+```bash
+$ kubectl get ns | grep conduit
+NAME                  STATUS    AGE
+conduit               Active    4m
+conduit-egress-test   Active    4m
+conduit-get-test      Active    3m
+...
+```
+
+## Cleanup
+
+To delete all of the test namespaces created while running the tests, run:
+
+```bash
+$ bin/test-cleanup <conduit-namespace>
+```
+
+## Writing tests
+
+To add a new test, create a new subdirectory inside the `test/` directory.
+Configuration files, such as Kubernetes configs, should be placed inside a
+`testdata/` directory inside the test subdirectory that you created. Then create
+a test file in the subdirectory that's suffixed with `_test.go`. This test file
+will be run automatically by the test runner script.
+
+The tests rely heavily on the test helpers that are defined in the `testutil/`
+directory. For a complete description of how to use the test helpers to write
+your own tests, view the `testutil` package's godoc, with:
+
+```bash
+$ godoc github.com/runconduit/conduit/testutil | less
+```
+
+# Integration tests: proxy-init
+
+The `proxy-init/` directory contains a separate set of integration tests, which
+can be run in your Kubernetes cluster. The instructions below assume that you
+are using [minikube](https://github.com/kubernetes/minikube).
+
+Start by building and tagging the `proxy-init` image required for the test:
+
+```bash
+DOCKER_TRACE=1 bin/mkube bin/docker-build-proxy-init
+bin/mkube docker tag gcr.io/runconduit/proxy-init:`bin/root-tag` gcr.io/runconduit/proxy-init:latest
+```
+
+The run the tests with:
+
+```bash
+cd proxy-init/integration_test
+eval $(minikube docker-env)
+./run_tests.sh
+```
+
+# Unit tests
```

Comment:
i may be biased as i'm so used to running these unit tests, but consider putting them first in the doc. i think as a new user to the repo (and perhaps making a specific change to one part of the codebase) the first thing i like to do is to run these types of tests to get a baseline that that area is building/working.